src/ROCrate/ROCrateObject.fs

```diff
@@ -0,0 +1,33 @@
+namespace ARCtrl.ROCrate
+
+open DynamicObj
+
+/// Base interface implemented by all explicitly known objects in our ROCrate profiles.
+type IROCrateObject =
+    abstract member SchemaType : string
+    abstract member Id: string
+    abstract member AdditionalType: string option
+
+/// Base class for all explicitly known objects in our ROCrate profiles to inherit from.
+/// Basically a DynamicObj that implements the IROPCrateObject interface.
+type ROCrateObject(id:string, schemaType: string, ?additionalType) =
+    inherit DynamicObj()
+
+    let mutable _schemaType = schemaType
+    let mutable _additionalType = additionalType
+
+    member this.Id 
```

Comment:
> Could you explain the decision to duplicate the properties exclusively on the type and in the interface? Is it because of ease of access

```fsharp
type A =
    abstract member B: string

type C() =
    interface A with 
        member this.B = "lol nah"

let lol = C()

lol.B // <- does not work
(lol :> A).B // <- must cast to access member B
```

Reply:
fixed